src/tools/debugger/dosdebug.c

```diff
@@ -517,10 +527,14 @@ int main (int argc, char **argv)
         }
         p[MHP_BUFFERSIZE - 1] = '\0';
         q = strpbrk(p, "\r\n");
-        *q = '\0';
-
-        handle_console_input(p); // p is freed by the function
+        if (q) {
+          *q = '\0';
+          q = p;
+        }
+        handle_console_input(q);
+        free(p);
```

Comment:
Now auto buffer is possible, no need for `free()`.

Reply:
Okay agreed. Will also change last_line back to static buffer.